boot.js

```diff
@@ -7,20 +7,26 @@ var EventEmitter = require('events')
 module.exports = function (cb) {
   var zenbot = { version }
   var args = minimist(process.argv.slice(3))
-  var conf
+  var conf = {}
 
+  // 1. load default config
+  try {
+    defaults = require('./conf')
```

Comment:
If `conf` isn't a _full_ copy of the `conf-sample` (for instance, if it only holds the values that a user wanted to customize), then it doesn't suffice as a full set of values for `defaults`, but the `conf-sample` will never be loaded.

Reply:
Indeed, thats why we tell people to make a copy of `conf-sample.js` as `conf.js` in the README
> cp conf-sample.js conf.js

My personal best practice usually goes like this:

1. Copy `conf-sample.js` to `conf.js`
2. Change `conf.js` (API keys, notifiers, etc.)
3. Create a custom conf (e.g. `cex_btc_usd.conf`) with period, neutral_rate, etc. for each instance to load with `--conf=...`

This way i have one base config and multiple custom configs to cover my separate instances on a single node.

IMHO sometimes its simpler to get people to "read (and follow) the f-ing manual" 😁 instead of trying to cover ever possible constellation they might come up with.

Reply:
That would work if we insist that users also update their `conf.js` every time a new option is added and a new default value is added to `conf-sample.js`, but I think that's somewhat unreasonable to ask.

Reply:
Yes, i think what better just merge it with default

Reply:
Fair point if the committer forgets to check if his/her new property actually exists.

> but I think that's somewhat unreasonable to ask

Not for me 😁 

Let me rework it.
brb

Reply:
Done.

I updated the merge comment to reflect the changes.